libopae++/include/opaec++/pvalue.h

```diff
@@ -66,29 +72,38 @@ struct guid_t {
   }
 
   void parse(const char *str) {
-    if (0 != uuid_parse(str, data_.data())) {
-      // throw error
+    int u;
+    if (0 != (u = uuid_parse(str, data_.data()))) {
+      log_.error() << "uuid_parse() failed with (" << u << ")";
+      throw except(OPAECXX_HERE);
     }
-    if (FPGA_OK != fpgaPropertiesSetGUID(*props_, data_.data())) {
-      // throw error
+    fpga_result res;
+    if (FPGA_OK != (res = fpgaPropertiesSetGUID(*props_, data_.data()))) {
+      log_.error() << "fpgaPropertiesSetGUID() failed with (" << res
+                   << ") " << fpgaErrStr(res);
+      throw except(res, OPAECXX_HERE);
     }
   }
 
   friend std::ostream &operator<<(std::ostream &ostr, const guid_t &g) {
     fpga_properties props = *g.props_;
     fpga_guid guid_value;
-    if (fpgaPropertiesGetGUID(props, &guid_value) == FPGA_OK) {
+    fpga_result res;
+    if ((res = fpgaPropertiesGetGUID(props, &guid_value)) == FPGA_OK) {
       char guid_str[84];
       uuid_unparse(g.data_.data(), guid_str);
       ostr << guid_str;
     } else {
-      // TODO: Log or throw
+      g.log_.error() << "fpgaPropertiesGetGUID() failed with (" << res
+                     << ") " << fpgaErrStr(res);
+      throw except(res, OPAECXX_HERE);
```

Comment:
What if the `res` variable is `FPGA_NOT_FOUND`?

Reply:
Ah, yes. Thanks